CHANGES.md

```diff
@@ -200,12 +200,22 @@ Misc
 
 SSReflect
 
+- New tactic `under` to rewrite under binders, given an extensionality lemma:
+  - interactive mode: `under lem`, associated terminator: `over`
+  - one-liner mode: `under lem do [tac1 | ...]`
+
+  It can take occurrence switches, contextual patterns, and intro-patterns:
```

Comment:
I would make this "intropattern" here and elsewhere for consistency with the rest of the doc.

Reply:
@jfehrle Thanks for your comments.

It seems `intropattern` is almost never used (except for Sphinx identifiers):
```
$ git grep -n -e "intropattern" --and --not -e '@intropattern\|_intropattern\|intropattern_\|`intropattern' "*.rst" | cat
doc/sphinx/proof-engine/tactics.rst:127:   intropattern            : *
doc/sphinx/proof-engine/tactics.rst:235:  If :tacn:`injection` is applicable, the intropattern
```

However, there are several occurrences of `intro-pattern` and `intro pattern`:
```
$ git grep -n -e "intro-pattern" "*.rst"
doc/sphinx/credits.rst:1077:   intro-patterns by Hugo Herbelin and others.
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1742:destructing intro-patterns.
doc/sphinx/proof-engine/ssreflect-proof-language.rst:3722:+ As a 2-branches intro-pattern is provided, it checks that 3 subgoals
doc/sphinx/proof-engine/ssreflect-proof-language.rst:3736:  intro-pattern branch (here, ``i``).
doc/sphinx/proof-engine/ssreflect-proof-language.rst:3738:If the intro-pattern is omitted, the Ltac expression:

$ git grep -n -e "intro pattern" "*.rst"
doc/sphinx/credits.rst:1452:  end of the intro pattern. In addition to that, the use-and-discard flag
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1506:   :n:`[: @ident ]` intro pattern (see section :ref:`introduction_ssr`).
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1577:The :token:`i_pattern`\s can be seen as a variant of *intro patterns*
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1614:from the :token:`i_view` with the ``-`` intro pattern or by putting
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1631:lets one write just ``/myop`` in the intro pattern. Note the scope
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1650:  from the :token:`ident` with the ``-`` intro pattern.
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1686:  of the intro pattern. For example ``move=> + y`` on a goal::
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1727:  does nothing, but counts as an intro pattern. It can also be used to
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1747:Clears are deferred until the end of the intro pattern.
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1763:If the cleared names are reused in the same intro pattern, a renaming
doc/sphinx/proof-engine/ssreflect-proof-language.rst:1806:  with :token:`ident`. If the intro pattern immediately follows a call
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2555:Another frequent use of the intro patterns combined with ``have`` is the
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2676:For this purpose the ``[: name ]`` intro pattern and the tactic
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2689:When intro patterns for abstract constants are used in conjunction
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2702:solved. Goals corresponding to intro patterns for abstract constants
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2718:Last, notice that the use of intro patterns for abstract constants is
doc/sphinx/proof-engine/ssreflect-proof-language.rst:2890:followed by a comma before the usual intro pattern. The identifier
doc/sphinx/proof-engine/ssreflect-proof-language.rst:5366:Note that the NO-OP intro pattern ``-`` can be used to separate two views,
doc/sphinx/proof-engine/ssreflect-proof-language.rst:5518:intro pattern :ref:`introduction_ssr`
doc/sphinx/proof-engine/tactics.rst:120:multiple hypotheses or subgoals.  Common tactics that accept intro patterns
doc/sphinx/proof-engine/tactics.rst:147:Note that the intro pattern syntax varies between tactics.
doc/sphinx/proof-engine/tactics.rst:183:Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
doc/sphinx/proof-engine/tactics.rst:264:   similar to |SSR|'s intro patterns.
doc/sphinx/proof-engine/tactics.rst:304:   .. example:: intro pattern for /\\
doc/sphinx/proof-engine/tactics.rst:320:   .. example:: intro pattern for \\/
doc/sphinx/proof-engine/tactics.rst:336:   .. example:: -> intro pattern
doc/sphinx/proof-engine/tactics.rst:352:   .. example:: [=] intro pattern
doc/sphinx/proof-engine/tactics.rst:377:   .. example:: (A & B & ...) intro pattern
doc/sphinx/proof-engine/tactics.rst:393:   .. example:: * intro pattern
doc/sphinx/proof-engine/tactics.rst:415:   .. example:: compound intro pattern
doc/sphinx/proof-engine/tactics.rst:428:   .. example:: combined intro pattern using [=] -> and %
doc/sphinx/proof-engine/tactics.rst:1082:   intro patterns.  See the description in :ref:`intropatterns`.
doc/sphinx/proof-engine/tactics.rst:1385:      If :n:`simple_intropattern` is an intro pattern (see :ref:`intropatterns`),
doc/sphinx/user-extensions/syntax-extensions.rst:1739:        - an intro pattern
```

so I propose to change the occurrences I had added in my patch to the most frequent form, `intro pattern`.

Reply:
I did that, and I also reworded the very last occurrence of `intro-pattern` to `intro pattern` in `doc/sphinx/proof-engine/ssreflect-proof-language.rst:1742`.

Reply:
Makes sense.  I didn't think to search for "intro pattern".